invoice/src/main/java/org/killbill/billing/invoice/dao/DefaultInvoiceDao.java

```diff
@@ -184,12 +184,24 @@ public List<InvoiceModelDao> getInvoicesByAccount(final Boolean includeVoidedInv
         });
     }
 
+    @Override
+    public Pagination<InvoiceModelDao> getPaginatedInvoicesByAccount(final Boolean includeVoidedInvoices, final Boolean includeInvoiceComponents, final Long offset, final Long limit, final InternalTenantContext context) {
+        final List<Tag> invoicesTags = getInvoicesTags(context);
+        return transactionalSqlDao.execute(true, entitySqlDaoWrapperFactory -> invoiceDaoHelper.getInvoicesByAccountFromTransactionPaginated(includeVoidedInvoices, includeInvoiceComponents, invoicesTags, entitySqlDaoWrapperFactory, offset, limit, paginationHelper, context));
```

Comment:
I think there is a problem with the approach: Most of our DAO methods are built using the `transactionalSqlDao.execute(true, wrapper)`, where the `wrapper` is of type `EntitySqlDaoTransactionWrapper`, i.e an interface whose implementation will run some code **meant to be run within a transaction**.

However, in your case, you pass `invoiceDaoHelper.getInvoicesByAccountFromTransactionPaginated(...)` which is not really a bunch of queries to be run in a transaction but rather a delegation to the pagination helper, which will itself [start a new transaction and run queries within the transaction](https://github.com/killbill/killbill/blob/work-for-release-0.23.x/util/src/main/java/org/killbill/billing/util/entity/dao/DefaultPaginationSqlDaoHelper.java#L70) (so we end up with two transactions one inside the other, and perhaps it ends up working but this is not the implementation we want).

I think we need to simplify and do something similar for other pagination dao apis, where we simply invoke the pagination helper, as shown [here](https://github.com/killbill/killbill/blob/work-for-release-0.23.x/payment/src/main/java/org/killbill/billing/payment/dao/DefaultPaymentDao.java#L607). The logic for the queries in embedded in the `PaginationIteratorBuilder#build` method.